server/server.go

```diff
@@ -43,6 +43,18 @@ func NewArtifactServer(path string, address string, timeout time.Duration) (*Art
 	return as, nil
 }
 
+func NewArtifactStore(c client.Client, scheme *runtime.Scheme, path, storageAddress string, artifactRetentionTTL time.Duration, artifactRetentionRecords int) (*storage.Storage, *ArtifactServer, error) {
```

Comment:
The storageAddress is not the same as the Advertised Storage address. The storage needs the Advertised address, but the artifact server needs the address which is with the localhost. This fails to start with the current address configuration.